src/vivarium/framework/randomness/index_map.py

```diff
@@ -215,6 +219,7 @@ def _convert_to_ten_digit_int(
         if pdt.is_datetime64_any_dtype(column):
             integers = self._clip_to_seconds(column.astype(np.int64))
         elif pdt.is_integer_dtype(column):
+            column = column.astype(int)
```

Comment:
This behavior is consistent with the other branch of the if/else block and makes mypy happy.